fast64_internal/f3d/f3d_material.py

```diff
@@ -3231,6 +3209,29 @@ class RDPSettings(PropertyGroup):
         update=update_node_values_with_preset,
     )
 
+    @property
+    def using_fog(self):
+        return self.g_fog or any(["FOG" in input for input in self.blend_inputs])
+
+    @property
+    def blend_color_inputs(self):
+        return (getattr(self, f"blend_{val}") for val in ("p1", "p2", "m1", "m2"))
+
+    @property
+    def blend_alpha_inputs(self):
+        return (getattr(self, f"blend_{val}") for val in ("a1", "a2", "b1", "b2"))
+
+    @property
+    def blend_inputs(self):
+        return self.blend_color_inputs
+        return self.blend_alpha_inputs
```

Comment:
Did you mean to do something like
```suggestion
        return self.blend_color_inputs + self.blend_alpha_inputs
```

Reply:
These are generators so I did mean that